### PR TITLE
Addressing API review feedback regarding json columns

### PR DESCRIPTION
**Pull request by maumar**
- renamed JsonColumnName to ContainerColumnName,
- renamed JsonColumnTypeMapping to ContainerColumnTypeMapping,
- changed RelationalModelValidator.ValidateJsonEntityProperties to protected method,
- moved JsonPropertyName methods on navigation into entity

### CODE REVIEW

File: src/EFCore.Relational/Extensions/RelationalEntityTypeBuilderExtensions.cs [hunk 1]
```diff
@@ -1146,4 +1146,43 @@ public static EntityTypeBuilder<TEntity> UseTptMappingStrategy<TEntity>(this Ent
             tableName,
             tableSchema,
             fromDataAnnotation ? ConfigurationSource.DataAnnotation : ConfigurationSource.Convention);
+
+    /// <summary>
+    ///     Configures the entity mapped to a JSON column, mapping it to a specific JSON property,
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Extensions/RelationalEntityTypeBuilderExtensions.cs:1151]:
```suggestion
    ///     Configures the entity mapped to a JSON column, mapping it to the given JSON property,
```

File: src/EFCore.Relational/Extensions/RelationalEntityTypeBuilderExtensions.cs [hunk 2]
```diff
@@ -1146,4 +1146,43 @@ public static EntityTypeBuilder<TEntity> UseTptMappingStrategy<TEntity>(this Ent
...
+            return null;
+        }
+
+        entityTypeBuilder.Metadata.SetJsonPropertyName(name, fromDataAnnotation);
+
+        return entityTypeBuilder;
+    }
+
+    /// <summary>
+    ///     Returns a value indicating whether the given value can be used as a JSON property name for a given navigation.
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Extensions/RelationalEntityTypeBuilderExtensions.cs:1177]:
```suggestion
    ///     Returns a value indicating whether the given value can be used as a JSON property name for the entity type.
```

File: src/EFCore.Relational/Metadata/Conventions/RelationalNavigationJsonPropertyNameAttributeConvention.cs [hunk 3]
```diff
@@ -42,7 +42,7 @@ public class RelationalNavigationJsonPropertyNameAttributeConvention :
     {
         if (!string.IsNullOrWhiteSpace(attribute.Name))
         {
-            navigationBuilder.HasJsonPropertyName(attribute.Name, fromDataAnnotation: true);
+            navigationBuilder.Metadata.TargetEntityType.Builder.HasJsonPropertyName(attribute.Name, fromDataAnnotation: true);
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Metadata/Conventions/RelationalNavigationJsonPropertyNameAttributeConvention.cs:45]: So the attribute is the only way of setting the `JsonPropertyName` right now? Can you file an issue to make `ToJson` set it?

**Review by AndriySvyryd — APPROVED** [review]

File: src/EFCore.Relational/Metadata/Conventions/RelationalNavigationJsonPropertyNameAttributeConvention.cs (hunk 3 above)
Comment (maumar) [reply src/EFCore.Relational/Metadata/Conventions/RelationalNavigationJsonPropertyNameAttributeConvention.cs:45]: you can use builder APIs also: 
`OwnedNavigationBuilder HasJsonPropertyName`
and 
`PropertyBuilder HasJsonPropertyName`

**Pull request merged**